docs/rules/no-global-assign.md

```diff
@@ -0,0 +1,47 @@
+# Disallow assignments to readonly global variables (no-global-assign)
+
+If people rewrite embedded global variables, it may cause severe problems.
```

Comment:
I'd suggest something like:

JavaScript environments contain a number of built-in global variables, such as `window` in browsers and `process` in Node.js. In almost all cases, you don't want to assign a value to these global variables as doing so could result in losing access to important functionality. For example, you probably don't want to do this in browser code:

```js
window = {};
```

While examples such as `window` are obvious, there are often hundreds of built-in global objects provided by JavaScript environments. It can be hard to know if you're assigning to a global variable or not.